File: Dynamic-Programming/72. Edit Distance/editDistance.py
class Solution:
    def minDistance(self, word1: str, word2: str) -> int:
        n = len(word1) + 1
        m = len(word2) + 1
        dp = [[0 for _ in range(n)] for _ in range(m)]
        
        for i in range(1, n):
            dp[0][i] = dp[0][i - 1] + 1
        
        for i in range(1, m):
            dp[i][0] = dp[i - 1][0] + 1
            
        for i in range(1, m):
            for j in range(1, n):
                if word1[j - 1] == word2[i - 1]:
                    dp[i][j] = dp[i - 1][j - 1]
                else:
                    dp[i][j] = min(dp[i - 1][j - 1], dp[i][j - 1], dp[i - 1][j]) + 1
                
        return dp[-1][-1]

File: Dynamic-Programming/72. Edit Distance/editDistance.js
/**
 * @param {string} word1
 * @param {string} word2
 * @return {number}
 */
 var minDistance = function(word1, word2) {
    let m = word1.length + 1;
    let n = word2.length + 1;
    let dp = new Array(n)
    for (let i = 0; i < n; i++) {
        dp[i] = new Array(m).fill(0);
    }
    
    for (let i = 1; i < m; i++) {
        dp[0][i] = dp[0][i - 1] + 1
    }
    
     for (let i = 1; i < n; i++) {
        dp[i][0] = dp[i - 1][0] + 1
    }
    
    for (let i = 1; i < n; i++) {
        for (let j = 1; j < m; j++) {
            if (word1[j - 1] === word2[i - 1]) {
                dp[i][j] = dp[i - 1][j - 1]
            } else {
                dp[i][j] = Math.min(dp[i - 1][j - 1], dp[i][j - 1], dp[i - 1][j]) + 1;
            }
        }
    }
    return dp[n - 1][m - 1];
};